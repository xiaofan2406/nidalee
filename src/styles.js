/* @flow */
import { css, keyframes } from 'react-emotion';

export const colors = {
  blue: '#349ef1',
  black: '#242729',
  white: '#ffffff',
  grey1: '#f5f5f5',
  grey2: '#e8e8e8',
  red1: '#fd3030',
  orange1: '#faad14',
  green1: '#23d160',
};

export const fontSizes = {
  small: 12,
  regular: 14,
  large: 18,
};

export const spacing = {
  unit: 6,
  internal: 8,
  external: 12,
  breath: 24,
  break: 36,
};

export const theme = {
  primaryColor: colors.blue,
  eleBgColor: colors.grey1,
  borderColor: colors.grey2,
  fontFamily: '"Open Sans", sans-serif',
  fontSize: fontSizes.regular,
  color: 'rgba(0, 0, 0, 0.75)',
  colorDanger: colors.red1,
  colorWarning: colors.orange1,
  colorSuccess: colors.green1,
};

export const defaultFont = css`
  font-family: 'Open Sans', sans-serif;
  font-size: 14px;
  line-height: 1.4;
`;

export const outline = css`
  outline: 1px solid ${theme.primaryColor};
`;

export const spinAnimation = keyframes`
  0% {
    transform: rotate(0deg)
  }
  100% {
    transform: rotate(360deg)
  }
`;

export const verticalScroll = css`
  overflow-y: scroll;
  &::-webkit-scrollbar {
    width: 10px;
    background-color: rgba(0, 0, 0, 0);
    border-radius: 40px;
  }
  &::-webkit-scrollbar:hover {
    background-color: rgba(0, 0, 0, 0.09);
  }
  &::-webkit-scrollbar-thumb:vertical {
    background-color: rgba(0, 0, 0, 0.5);
    border-radius: 40px;
    background-clip: padding-box;
    border: 2px solid rgba(0, 0, 0, 0);
    min-height: 10px;
  }
  &::-webkit-scrollbar-thumb:vertical:active {
    background-color: rgba(0, 0, 0, 0.61);
  }
`;
