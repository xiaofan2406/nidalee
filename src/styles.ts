import { css, keyframes } from '@emotion/core';

export const colors = {
  white: '#ffffff',
  blue1: '#349ef1',
  blue2: '#008fff',
  black1: '#131313',
  black2: '#232323',
  black3: '#333333',
  black4: '#434343',
  black5: '#4f4f4f',
  red1: '#fd3030',
  orange1: '#faad14',
  green1: '#23d160',
};

export const fontSizes: Record<Size, number> = {
  small: 12,
  regular: 14,
  large: 18,
};

export const theme = {
  textColor: 'rgba(0,0,0,0.85)',
  textColorInverse: colors.white,

  subTextColor: 'rgba(0,0,0,0.3)',
  fontSize: fontSizes.regular,
  fontFamily: '"Open Sans", sans-serif',

  eleBgColor: colors.black3,

  primaryColor: colors.blue2,
  colorDanger: colors.red1,
  colorWarning: colors.orange1,
  colorSuccess: colors.green1,
};

export const baseContainer = css`
  padding: 12px 16px;
`;

export const baseArea = css`
  padding: 8px 12px;
`;

export const defaultFont = css`
  font-family: ${theme.fontFamily};
  font-size: ${theme.fontSize}px;
  color: ${theme.textColor};
  line-height: 1.4;
`;

export const outline = css`
  outline: 1px solid ${theme.primaryColor};
`;

export const spinAnimation = keyframes`
  0% {
    transform: rotate(0deg)
  }
  100% {
    transform: rotate(360deg)
  }
`;

export const verticalScroll = css`
  overflow-y: scroll;
  &::-webkit-scrollbar {
    width: 10px;
    background-color: rgba(0, 0, 0, 0);
    border-radius: 40px;
  }
  &::-webkit-scrollbar:hover {
    background-color: rgba(0, 0, 0, 0.09);
  }
  &::-webkit-scrollbar-thumb:vertical {
    background-color: rgba(0, 0, 0, 0.5);
    border-radius: 40px;
    background-clip: padding-box;
    border: 2px solid rgba(0, 0, 0, 0);
    min-height: 10px;
  }
  &::-webkit-scrollbar-thumb:vertical:active {
    background-color: rgba(0, 0, 0, 0.61);
  }
`;

// remove firefox dotted outline
export const noFirefoxOutline = css`
  &::-moz-focus-inner {
    border: 0;
  }
`;
