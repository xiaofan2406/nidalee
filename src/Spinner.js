/* @flow */
import React from 'react';
import styled from 'react-emotion';
import { theme, spinAnimation } from './styles';

const SpinnerDiv = styled.div`
  display: inline-block;
  & > div {
    width: ${({ size }) => size}px;
    height: ${({ size }) => size}px;
    border: ${({ size }) => size / 8.5}px solid ${({ color }) => color};
    border-bottom-color: transparent;
    border-radius: 100%;
    background: transparent;

    animation: ${spinAnimation} 0.75s 0s infinite linear;
    animation-fill-mode: both;
  }
`;

type SpinnerProps = {
  size?: number,
  color?: string,
};

const Spinner = (props: SpinnerProps) => (
  <SpinnerDiv {...props}>
    <div />
  </SpinnerDiv>
);

Spinner.defaultProps = {
  size: 60,
  color: theme.primaryColor,
};

export default Spinner;
