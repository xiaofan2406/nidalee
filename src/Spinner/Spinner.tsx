import { css } from '@emotion/core';
import React, { SFC } from 'react';
import { spinAnimation, theme } from '../styles';

export interface SpinnerProps extends React.HTMLAttributes<HTMLDivElement> {
  /**
   * The width and height for the spinner.
   */
  size?: number;
  /**
   * The thickness scale for the spinner.
   */
  scale?: number;
  /**
   * The color for the spinner.
   */
  color?: string;
}

const Spinner: SFC<SpinnerProps> = ({
  size,
  scale,
  color,
  className,
  ...rest
}) => {
  const maxBorder = size! / 2;
  const scaledBorder = (size! * scale!) / 10;
  const borderSize = scaledBorder >= maxBorder ? maxBorder : scaledBorder;

  return (
    <div
      css={css`
        display: inline-block;
      `}
      {...rest}
    >
      <div
        css={css`
          width: ${size!}px;
          height: ${size!}px;
          border: ${borderSize}px solid ${color};
          border-bottom-color: transparent;
          border-radius: 100%;
          background: transparent;
          animation: ${spinAnimation} 0.75s 0s infinite linear both;
        `}
      />
    </div>
  );
};

Spinner.defaultProps = {
  size: 60,
  scale: 1,
  color: theme.primaryColor,
};

export default Spinner;
