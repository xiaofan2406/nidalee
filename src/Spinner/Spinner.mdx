---
name: Spinner
---

import { Playground, PropsTable } from 'docz'
import Spinner from './Spinner'

# Spinner

## Props

<PropsTable of={Spinner} />

## Basic usage

<Playground>
  <Spinner />
  <Spinner size={120} />
  <Spinner size={120} scale={0.5} />
</Playground>
