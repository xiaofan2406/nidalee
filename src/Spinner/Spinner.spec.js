import React from 'react';
import { render } from 'react-testing-library';
import Spinner from './Spinner';
import { theme } from '../styles';

test('snapshot matches', () => {
  const { container } = render(<Spinner />);
  expect(container.firstChild).toMatchSnapshot();
});

test('the border width will not exceed half of the size', () => {
  const { container } = render(<Spinner size={120} scale={7} />);
  expect(container.firstChild.firstChild).toHaveStyleRule(
    'border',
    `60px solid ${theme.primaryColor}`
  );
});
