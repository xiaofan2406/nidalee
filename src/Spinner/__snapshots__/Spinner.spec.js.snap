// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot matches 1`] = `
<div
  class="css-13z7859-cssSpinner"
>
  <div
    class="css-1s3rwo4-Spinner"
  />
</div>
`;
